### PR TITLE
Add index file option for distributedLoad command

### LINKED ISSUE #12553: If alluxio can support index file based distributedLoad CLI
**Is your feature request related to a problem? Please describe.**
Given a UFS is mounted, we only want to distributedLoad a subset of files which are listed in a index file.

**Describe the solution you'd like**
an index file can be provided in distributedLoad CLI to specify files being loaded.

**Describe alternatives you've considered**
N/A

**Urgency**
Medium. 

**Additional context**

### PR DESCRIPTION
**Pull request by jenoudet**
Close #12553

### CODE REVIEW

File: shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java [hunk 1]
```diff
@@ -109,7 +120,20 @@ public int run(CommandLine cl) throws AlluxioException, IOException {
     String[] args = cl.getArgs();
     AlluxioURI path = new AlluxioURI(args[0]);
     int replication = FileSystemShellUtils.getIntArg(cl, REPLICATION_OPTION, DEFAULT_REPLICATION);
-    distributedLoad(path, replication);
+
+    if (!cl.hasOption('f')) {
+      distributedLoad(path, replication);
+    } else {
+      FileInStream is = mFileSystem.openFile(path);
```
Comment (yuzhu) [inline shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java:127]: did we decide to use local file system for the index file path? if  that is the case, perhaps use `BufferedReader` ? something like `BufferedReader br = new BufferedReader(path);`

File: tests/src/test/java/alluxio/client/cli/fs/command/DistributedLoadCommandTest.java [hunk 2]
```diff
@@ -60,4 +64,28 @@ public void loadFile() throws IOException, AlluxioException {
...
     Assert.assertEquals(100, status.getInMemoryPercentage());
   }
+
+  @Test
+  public void loadIndexFile() throws IOException, AlluxioException {
+    FileSystemTestUtils.createByteFile(sFileSystem, "/testFileA", WritePType.THROUGH, 10);
+    FileSystemTestUtils.createByteFile(sFileSystem, "/testFileB", WritePType.THROUGH, 10);
+
+    AlluxioURI uri = new AlluxioURI("/testFile");
+    FileOutStream out = sFileSystem.createFile(uri);
```
Comment (yuzhu) [inline tests/src/test/java/alluxio/client/cli/fs/command/DistributedLoadCommandTest.java:74]: Similarly, could we use a local file system path for the index file?

**Review by yuzhu — CHANGES_REQUESTED** [review]
@jenoudet Thanks for the fix! Left a few comments and also could you fix the checkstyle? 

Thanks.

File: tests/src/test/java/alluxio/client/cli/fs/command/DistributedLoadCommandTest.java [hunk 3]
```diff
@@ -60,4 +63,28 @@ public void loadFile() throws IOException, AlluxioException {
     status = sFileSystem.getStatus(uri);
     Assert.assertEquals(100, status.getInMemoryPercentage());
   }
+
+  @Test
+  public void loadIndexFile() throws IOException, AlluxioException {
+    FileSystemTestUtils.createByteFile(sFileSystem, "/testFileA", WritePType.THROUGH, 10);
+    FileSystemTestUtils.createByteFile(sFileSystem, "/testFileB", WritePType.THROUGH, 10);
+
+    String dir = System.getProperty("user.dir");
```
Comment (yuzhu) [inline tests/src/test/java/alluxio/client/cli/fs/command/DistributedLoadCommandTest.java:72]: Consider using a `TemporaryFolder` provided by JUnit? 
```  
@Rule
  public TemporaryFolder mTestFolder = new TemporaryFolder();
```

**Review by yuzhu — CHANGES_REQUESTED** [review]
@jenoudet Overall looks good, just one minor comment.

**Review by yuzhu — APPROVED** [review]
LGTM. thanks

File: shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java [hunk 4]
```diff
@@ -65,6 +67,15 @@
                   + "The default upper limit is "
                   + AbstractDistributedJobCommand.DEFAULT_ACTIVE_JOBS)
           .build();
+  private static final Option INDEX_FILE =
+      Option.builder("f")
+          .required(false)
```
Comment (apc999) [inline shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java:72]: how about
```
      Option.builder()
          .longOpt("index")
```

**Comment by apc999** [comment]
@jenoudet one minor comment left on the option name. I prefer to use `--index` and leave `-f` for some other future use.
Otherwise, this looks good to me. Thanks!

**Review by apc999 — APPROVED** [review]
Thanks for your contribution!

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**